### ISSUE TITLE
feat(developer): Compile keyboards w/ new features with min v15

### ISSUE THREAD

**Issue by darcywong00** [keymanapp/keyman#5992]
#5894 added Developer support for U_xxxx_yyyy keys. However, versions of Keyman for Android 14.0 and earlier will crash if attempting to render those subkeys.

Per @mcdurdin the compiler  should tag those keyboards as min v15.
Also applies for NewContext, PostKeystroke, &LayerChanged, and readonly group.

I'd need to check if Keyman for Android will prevent installing packages with versions > than itself.

**Comment by mcdurdin** [comment]
> Also applies for NewContext, PostKeystroke, &LayerChanged, and readonly group.

I think we can get away with not requiring v15 for these -- there will just be a graceful loss of functionality without any bad side effects. (The only one I was worried about, &LayerChanged, is only useful in PostKeystroke group processing anyway, so it's not an issue.)

**Comment by darcywong00** [comment]
> I'd need to check if Keyman for Android will prevent installing packages with versions > than itself.

I'm not finding minimum Keyman version in kmp.json,
```json
  "system": {
    "keymanDeveloperVersion": "14.0.274.0",
    "fileVersion": "7.0"
  },
```

How should I proceed in preventing Keyman for Android 14.0 subkeys from crashing on U_xxxx_yyyy?

**Comment by darcywong00** [comment]
@mcdurdin 
I built a test keyboard with Developer 15 that has a longpress of U_xxxx_yyyy on a key. kmp.json has
```
{
  "system": {
    "keymanDeveloperVersion": "15.0.146.0",
    "fileVersion": "7.0"
  },
```

Should fileVersion be 15.0 for Keyman 15 features?

**Comment by mcdurdin** [comment]
> @mcdurdin I built a test keyboard with Developer 15 that has a longpress of U_xxxx_yyyy on a key. kmp.json has
> 
> ```
>     "fileVersion": "7.0"
> ```
> 
> Should fileVersion be 15.0 for Keyman 15 features?

Yes, with #6145, the `fileVersion` will be `"15.0"`. From the testing in that PR, we can see that we have to fix several platforms:

1. Windows - OK
2. macOS - Silent failure to install? (@sgschantz)
3. Android - Ignores fileVersion (@darcywong00)
4. iOS - Ignores fileVersion (@sgschantz)
5. Linux - Unknown? (@ermshiperete)

I think we should try and fix this today for better file safety going forward. Linux and macOS *should* test the .kmx version anyway at runtime, but it really should be blocked at install time.
